rpcs3/rpcs3qt/gui_settings.cpp

```diff
@@ -165,34 +168,49 @@ void gui_settings::ShowBox(QMessageBox::Icon icon, const QString& title, const Q
 		return;
 	}
 
-	const QFlags<QMessageBox::StandardButton> buttons = icon != QMessageBox::Information ? QMessageBox::Yes | QMessageBox::No : QMessageBox::Ok;
-
-	QMessageBox* mb = new QMessageBox(icon, title, text, buttons, parent, Qt::Dialog | Qt::MSWindowsFixedSizeDialogHint | (always_on_top ? Qt::WindowStaysOnTopHint : Qt::Widget));
-	mb->deleteLater();
+	QMessageBox* mb = new QMessageBox(icon, title, text, {}, parent, Qt::Dialog | Qt::MSWindowsFixedSizeDialogHint | (always_on_top ? Qt::WindowStaysOnTopHint : Qt::Widget));
```

Comment:
This function is not supposed to know about different gui_saves, so why add installation specifics here?

Reply:
Avoiding duplication of code with GUI settings interaction.

Reply:
There are multiple bad decisions here...
- You move from a simple QMessageBox::StandardButton implementation to a custom implementation that regresses the existing code and also adds more unnecessary translation
- You add a specific firmware install callback into this method

So my suggestion:
- Check the result param after calling ShowBox to call the Emu callback. This removes the nasty hack.
- Either use the ib_confirm_fw entry or not. And if you use it, don't add code based on entry.... you may add a new buttons param that overrides the existing Standardbuttons. Remember, there's already an ::Ignore button and you can word your dialog in a way that makes sense without using "Install".
- mb->exec() should already return the chosen result if you don't add custom buttons to this generic dialog. So you don't have to check some clickedButton

Reply:
Fixed.